### PR TITLE
Priors in new storage

### LINKED ISSUE #770: Extract and store the the theoretical prior for a parameter in the new storage

### LINKED ISSUE #771: Expose the theoretical prior for a parameter from the new storage

### PR DESCRIPTION
**Pull request by mortalisk**
Resolves #770
Resolves #771 

Extracts priors and exposes them in http api

Depends on a PR in libres, https://github.com/equinor/libres/pull/927

### CODE REVIEW

File: ert_shared/storage/model.py [hunk 1]
```diff
@@ -252,3 +252,17 @@ class ErtBlob(Blobs):
...
+prior_ensemble_association_table = Table('prior_ensemble_association_table', Entities.metadata,
+    Column('prior_id', String, ForeignKey('parameter_priors.key')),
+    Column('ensemble_id', Integer, ForeignKey('ensembles.id'))
+)
+
+class ParameterPrior(Entities):
+    __tablename__ = "parameter_priors"
+
+    key = Column('key', String, primary_key=True)
+    data = Column('data', JSON)
```
Comment (jondequinor) [inline ert_shared/storage/model.py:267]: This kills reverse indexing. Do you foresee that as a problem?

File: ert_shared/storage/model.py [hunk 2]
```diff
@@ -252,3 +252,17 @@ class ErtBlob(Blobs):
...
+
+prior_ensemble_association_table = Table('prior_ensemble_association_table', Entities.metadata,
+    Column('prior_id', String, ForeignKey('parameter_priors.key')),
+    Column('ensemble_id', Integer, ForeignKey('ensembles.id'))
+)
+
+class ParameterPrior(Entities):
+    __tablename__ = "parameter_priors"
+
+    key = Column('key', String, primary_key=True)
```
Comment (sondreso) [inline ert_shared/storage/model.py:266]: Should there not be an association between the ParameterDefinition that is sampled from a Prior other than the implicit key link?

File: ert_shared/storage/model.py (hunk 1 above)
Comment (oysteoh) [reply ert_shared/storage/model.py:267]: i usually not want my primary keys exposed, but that is my opinion..

Comment (mortalisk) [reply ert_shared/storage/model.py:267]: @jondequinor not sure if we need that. Did it the simplest way. If we are to make more of it I think it should be discussed in the squad. 
@oysteoh what do you mean by primary keys?

File: ert_shared/storage/model.py (hunk 2 above)
Comment (mortalisk) [reply ert_shared/storage/model.py:266]: There is another implicit association also, through the ensemble it belongs to. The root ensemble will have the prior explicitly. Not sure if that is enough either though. What do you think? Perhaps we should just get it in as it is now, and create more issues to discuss how it should be?

File: ert_shared/storage/model.py (hunk 1 above)
Comment (oysteoh) [reply ert_shared/storage/model.py:267]: you have added key as a primary key. i don't want to expose my primary keys from the database out in the public. usually i also don't want strings as primary keys - i prefer incremental int's.

**Pull request merged**